pkg/controller/podautoscaler/metrics/metrics_client_test.go

```diff
@@ -57,9 +57,13 @@ func makeTestServer(t *testing.T, responses map[string]*serverResponse) (*httpte
 	mux := http.NewServeMux()
 
 	mkHandler := func(url string, response serverResponse) *util.FakeHandler {
+		codec, err := testapi.GetCodecForObject(response.obj.(runtime.Object))
```

Comment:
It's because the response.obj in this test sometimes is v1 object, like pod, (https://github.com/caesarxuchao/kubernetes/blob/move-v1alpha1/pkg/controller/podautoscaler/metrics/metrics_client_test.go#L108)